src/Components/DataHistoryExplorer/Internal/TimeSeriesBuilder/TimeSeriesBuilder.tsx

```diff
@@ -0,0 +1,227 @@
+
+    // hooks
+    const { t } = useTranslation();
+    const addTimeSeriesTwinCalloutId = useId('add-time-series-twin-callout');
+
+    // callbacks
+    const handleAddNew = useCallback(() => {
+        setIsTimeSeriesTwinCalloutVisible(true);
+    }, []);
+    const handleTimeSeriesTwinCalloutPrimaryAction = useCallback(
+        (timeSeriesTwin: IDataHistoryTimeSeriesTwin) => {
+            setTimeSeriesTwins(
+                produce(timeSeriesTwins, (draft) => {
+                    if (selectedTimeSeriesTwinId) {
+                        const selectedIdx = timeSeriesTwins.findIndex(
+                            (tsTwin) =>
+                                tsTwin.twinId === selectedTimeSeriesTwinId
+                        );
+                        draft[selectedIdx] = timeSeriesTwin;
+                    } else {
+                        timeSeriesTwin.chartProps.color = getHighChartColor(
+                            timeSeriesTwins.length
+                        );
+                        draft.push(timeSeriesTwin);
+                    }
+                })
+            );
+            setIsTimeSeriesTwinCalloutVisible(false);
+            setSelectedTimeSeriesTwinId(null);
+        },
+        [timeSeriesTwins, selectedTimeSeriesTwinId]
+    );
+    const onTimeSeriesTwinEdit = useCallback((twinId: string) => {
+        setSelectedTimeSeriesTwinId(twinId);
+        setIsTimeSeriesTwinCalloutVisible(true);
+    }, []);
+    const onTimeSeriesTwinRemove = useCallback(
+        (twinId: string) => {
+            setSelectedTimeSeriesTwinId(null);
+            setTimeSeriesTwins(
+                produce(timeSeriesTwins, (draft) => {
+                    const selectedIdx = draft.findIndex(
+                        (tsTwin) => tsTwin.twinId === twinId
+                    );
+                    draft.splice(selectedIdx, 1);
+                })
+            );
+        },
+        [timeSeriesTwins]
+    );
+
+    // side effects
+    useEffect(() => {
+        onTimeSeriesTwinListChange?.(timeSeriesTwins);
+    }, [timeSeriesTwins]);
+    const timeSeriesTwinList = useMemo(
+        () =>
+            getTimeSeriesTwinListItems(
+                timeSeriesTwins,
+                onTimeSeriesTwinEdit,
+                onTimeSeriesTwinRemove,
+                t
+            ),
+        [timeSeriesTwins, onTimeSeriesTwinEdit, onTimeSeriesTwinRemove, t]
+    );
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    return (
+        <div className={classNames.root}>
+            <Stack tokens={{ childrenGap: 8 }}>
+                <h2 className={classNames.header}>
+                    {t('dataHistoryExplorer.builder.title')}
+                </h2>
+                <span className={classNames.description}>
+                    {t('dataHistoryExplorer.builder.description')}
+                </span>
+            </Stack>
+            <Separator />
+            <CardboardList
```

Comment:
In the event that we cast, I think we should make the type numeric in the List

![image](https://user-images.githubusercontent.com/4854357/210450845-3fe69330-9c35-43fe-a956-817845b306b7.png)